scipy/special/special_c_wrappers.h

```diff
@@ -0,0 +1,54 @@
+#pragma once
+
```

Comment:
I think this file is a great idea. In a future PR, we should combine all the various C wrappers together, and can use something like "special_XXXX" as the naming style. Just a thin wrapper that calls out to the appropriate C++, like you have here.